tests/lib/nested.sh

```diff
@@ -64,18 +88,82 @@ create_nested_core_vm(){
     fi
 }
 
-start_nested_vm(){
-    systemd_create_and_start_unit nested-vm "${QEMU} -m 2048 -nographic -net nic,model=virtio -net user,hostfwd=tcp::8022-:22 -drive file=/tmp/work-dir/ubuntu-core.img,if=virtio,cache=none,format=raw -drive file=${PWD}/assertions.disk,if=virtio,cache=none,format=raw"
+start_nested_core_vm(){
+    systemd_create_and_start_unit nested-vm "${QEMU} -m 2048 -nographic -net nic,model=virtio -net user,hostfwd=tcp::$SSH_PORT-:22 -drive file=$WORK_DIR/ubuntu-core.img,if=virtio,cache=none,format=raw -drive file=${PWD}/assertions.disk,if=virtio,cache=none,format=raw -monitor tcp:127.0.0.1:$MON_PORT,server,nowait -usb"
     if ! wait_for_ssh; then
         systemctl restart nested-vm
     fi
 }
 
-destroy_nested_core_vm(){
+create_nested_classic_vm(){
+    mkdir -p "$WORK_DIR"
+
+    # Get the cloud image
+    IMAGE_URL=$(get_image_url_for_nested_vm)
+    wget -P "$WORK_DIR" "$IMAGE_URL"
+    IMAGE=$(ls $WORK_DIR/*.img)
+
+    # Prepare the cloud-init configuration
+    cat <<EOF > "$WORK_DIR/seed"
+#cloud-config
+  ssh_pwauth: True
+  users:
+   - name: user1
+     sudo: ALL=(ALL) NOPASSWD:ALL
+     shell: /bin/bash
+  chpasswd:
+   list: |
+    user1:ubuntu
+   expire: False
+EOF
+    cloud-localds -H "$(hostname)" "$WORK_DIR/seed.img" "$WORK_DIR/seed"
+
+    # Start the vm
+    start_nested_classic_vm "$IMAGE"
+}
+
+start_nested_classic_vm(){
+    IMAGE=$1
+    QEMU=$(get_qemu_for_nested_vm)
+
+    systemd_create_and_start_unit nested-vm "${QEMU} -m 2048 -nographic -net nic,model=virtio -net user,hostfwd=tcp::$SSH_PORT-:22 -hda $IMAGE -hdb $WORK_DIR/seed.img -monitor tcp:127.0.0.1:$MON_PORT,server,nowait -usb"
+    wait_for_ssh
+}
+
+destroy_nested_vm(){
     systemd_stop_and_destroy_unit nested-vm
-    rm -rf /tmp/work-dir
+    rm -rf "$WORK_DIR"
 }
 
 execute_remote(){
-    sshpass -p ubuntu ssh -p 8022 -o ConnectTimeout=10 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no user1@localhost "$*"
-}   
+    sshpass -p ubuntu ssh -p "$SSH_PORT" -o ConnectTimeout=10 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no user1@localhost "$*"
+}
+
+copy_remote(){
+    sshpass -p ubuntu scp -P "$SSH_PORT" -o ConnectTimeout=10 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no "$*" user1@localhost:~
+}
+
+add_usb_tablet_device(){
```

Comment:
Thank you! I love the idea of such helpers, very neat.

I think these particular devices will not be useful in hotplug tests, we will not have interfaces handling them anytime soon. The first target for hotplug tests is going to be a USB serial port adapter , e.g. `-chardev soc
ket,id=usbchr0,port=9022,server,host=localhost,nowait -usb -device usb-serial,chardev=usbchr0` (this is something I played around, it creates a virtual serial port with a local socket to use for read/writes).